### PR TITLE
Add javascript for upload

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: npm/src/auth/index.ts [hunk 1]
```diff
@@ -17,3 +21,33 @@ export const getKeycloakInstance: () => Promise<
...
+  keycloak: Keycloak.KeycloakInstance<"native">
+) => Promise<string> = async keycloak => {
+  const token = await refreshOrReturnToken(keycloak)
+  const credentials = new CognitoIdentityCredentials({
+    IdentityPoolId: TDR_IDENTITY_POOL_ID,
+    Logins: {
+      [TDR_IDENTITY_PROVIDER_NAME]: token
+    }
+  })
+  AWS.config.update({ region: "eu-west-2", credentials })
```
Comment (TomJKing) [inline npm/src/auth/index.ts:49]: Maybe region should be set as env variable?

File: npm/src/s3upload/index.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    tdrFile,
+    callback,
+    progressInfo
+  ) => {
+    const { file, fileId } = tdrFile
+    console.log(`${this.identityId}/${fileId}`)
+    const progress: S3.ManagedUpload = this.s3.upload({
+      Key: `${this.identityId}/${fileId}`,
+      Body: file,
+      Bucket: "tdr-upload-files-intg"
```
Comment (TomJKing) [inline npm/src/s3upload/index.ts:39]: Should the bucket name be hardcoded? Would be different depending on the environment

File: npm/src/auth/index.ts (hunk 1 above)
Comment (MancunianSam) [reply npm/src/auth/index.ts:49]: I've set it as a webpack variable but then hard coded it into the package json. I don't think we'll ever need to change it but if we do, I can set it in the Jenkins container.

File: npm/src/s3upload/index.ts (hunk 2 above)
Comment (MancunianSam) [reply npm/src/s3upload/index.ts:39]: Good spot, I've added the stage variable and used it here.

**Pull request merged**